sdk/python/kfp/dsl/pipeline_task_base.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pipeline task ABC.
+
+Used for compiled pipeline tasks and local pipeline tasks.
+"""
+
+import abc
+from typing import Any, Dict, List, Optional
+
+
+class PipelineTaskBase(abc.ABC):
```

Comment:
Admittedly I haven't gave enough thoughts on the design choice here, but it feels to me this might not be the right abstraction. Isn't `task` an entity that could be executable in one or more environments. For a task that's executable both locally and remotely, is there a necessity to split it into `PipelineTask` vs. `LocalTask`?